rfcs/text/responsiveGrid.md

```diff
@@ -0,0 +1,138 @@
+the `<GridLayout />` will supply props to define breakpoints and number of columns and will be styled using CSS Grid Layout.
+
+Code example
+```
+  import AppWrapper from 'carbon-react/lib/components/app-wrapper';
+  import { GridLayout, GridRow, GridColumn }  from 'carbon-react/lib/components/grid';
+  
+  const Index = () => (
+  <AppWrapper>
+    <GridLayout 
+      breakpoints={[100, 200, 300, 400, 500]}
+      columns={12}>
+        <GridRow>
+            <GridColumn>
+              <MyComponent />
+            </GridColumn>
+            <GridColumn>
+              <MyComponent /> 
+            </GridColumn>
+            <GridColumn>
+              <MyComponent />
+            </GridColumn>
+        </GridRow>
+      </GridLayout>
+    </AppWrapper>
+  );
+```
+
+## CSS Grid Layout
+CSS Grid Layout is a truly two dimensional grid allowing us to specify dimensions of rows and columns and control where we position components within the grid.
+
+Example
+```
+.parent {
+  display: grid;
+  grid-template-columns: 1fr 3fr 1fr; /* Three columns, one three times as wide as the others */
+  grid-template-rows: 200px auto 100px; /* Three rows, two with explicit widths */
+  grid-template-areas:
+    "header header header"
+    ". main sidebar"
+    "footer . .";
+}
+
+/*
+  Now, we can explicitly place items in the defined rows and columns.
+*/
+.child-1 {
+  grid-area: header;
+}
+
+.child-2 {
+  grid-area: main;
+}
+
+.child-3 {
+  grid-area: sidebar;
+}
+
+.child-4 {
+  grid-area: footer;
+}
+```
+# Drawbacks
+
+When you consider that a grid layout is created to serve the same content on different view ports it comes with a few disadvantage:
+
+1. Development time. It requires more work from the designer and developer than the equivalent single width design. Although this is true, familiarity will increase the turn around speed.
+2. Page load speed. All the data is served to all view ports, regardless of how much real estate you have available and how much content is actually displayed.
+3. Accessibility. Grid position of a component may be different to the DOM order causing flow and context discrepancies when using screen readers, and users who navigate with a keyboard may experience unexpected sequences when tabbing through components. The recommendation is that grid ordering and placement is used for only for visual, not logical, reordering of content. 
+
+For an explaination of what this means in practice see the [Mozilla CSS Grid Layout](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/CSS_Grid_Layout_and_Accessibility#Visual_not_logical_re-ordering)
+
+
+# Alternatives
+
+## CSS Flexbox
+Flexbox is designed for layout in one dimension (either a row or a column), although it does enable wrapping you can't control where components will be positioned since they are really just being pushed along a single axis until they wrap.
```

Comment:
```suggestion
Flexbox is designed for layout in one dimension (either a row or a column). Although it does support wrapping, the second-dimension positioning of components cannot be controlled (since the components are just being pushed along a single axis until they wrap).
```